language/tools/move-package/src/resolution/resolution_graph.rs

```diff
@@ -565,6 +566,29 @@ impl ResolvingGraph {
                             dep_name
                         )
                     })?;
+            } else {
```

Comment:
I think we can make this a bit lazier--if `git_rev` is a hash instead of a branch, updates aren't possible and we can skip the fetch.